### ISSUE TITLE
Trying to use apt-get, and fails update, upgrade and install build-essential

### ISSUE THREAD

**Issue by ahrensd** [Microsoft/WSL#2645]
Window ver:
Microsoft Windows [Version 10.0.15063] 
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.2 LTS
Release:        16.04
Codename:       xenial

Can't update the system using apt-get, always get missing files or failed to stat errors.
did take a while to get pass the corporate proxy (using fiddler and setting http_proxy environment variable and including 
`Acquire::http::Proxy "http://127.0.0.1:8888";
Acquire::https::Proxy "http://127.0.0.1:8888";
Acquire::ForceIPv4 "true";`
in /etc/apt/apt.conf.d/80Proxy

have run the commands several times with the output in the following files:-
[bash_out.txt](https://github.com/Microsoft/WSL/files/1456547/bash_out.txt)
[bash_build.txt](https://github.com/Microsoft/WSL/files/1456564/bash_build.txt)

Then tried use the recommended strace command and have attached the output:-
[bash_strace.txt](https://github.com/Microsoft/WSL/files/1456605/bash_strace.txt)
[strace.zip](https://github.com/Microsoft/WSL/files/1456606/strace.zip)

Thanks for looking into this....

**Comment by DarthSpock** [comment]
@ahrensd Have you tried upgrading to Build 16299.15 (Fall Creator's Update). Alot of the network issues were resolved with this release. If you go Insider, you should see a few more fixes as well as described in https://msdn.microsoft.com/en-us/commandline/wsl/release-notes

**Comment by ahrensd** [comment]
On a corporate build of Windows 10, and already using the early adopters version. So have no control on which Windows 10 version or upgrades.
Seems to be close to working. the networking appears to be working as the packages are downloaded. 
Hoping for some insight into a work around, someone that had and solved this problem, before!
Any ideas what to 'Failed to stat' errors are?

**Comment by ahrensd** [comment]
Was trying some things out and got apt-get to suddenly work
command I tried
`apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80
apt-key update
sudo apt-key update
rm -R /tmp/*
sudo rm -R /tmp/*
--> also attemped to remove files using windows explorer (file will not go using either method)
sudo apt-key update
sudo apt-key list
apt-get update`

And it worked, installed all sorts of stuff successfully.
Exited the bash shell.

Go to do a apt-get upgrade (or anything, and back to the same old problem)
Ofcourse I've tried to above commands and no luck. So I don't know what is going on.
the files are downloaded correctly and stored in /var/cache/apt/archives/partial
but get the following error at the end of the process 
`E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/v/vlan/vlan_1.9-3.2ubuntu1.16.04.4_amd64.deb  rename failed, No such file or directory (/var/cache/apt/archives/partial/vlan_1.9-3.2ubuntu1.16.04.4_amd64.deb -> /var/cache/apt/archives/vlan_1.9-3.2ubuntu1.16.04.4_amd64.deb).`
File is defitely there....

**Comment by DarthSpock** [comment]
Apt-get upgrade fails for me as well with 17.10 Artful but instead of the vlan .deb failing, it's the bash that's failing. It was odd because when I went to the /var/cache/apt/archives directory and performed an ls, it was still there. I tried gzip to re-extract it to see if it was a corruption from the upgrade from 16.04 to 17.10 and that failed. I went so far as to search the Ubuntu webpages to find the bash .deb file and removed the old file and downloaded using Windows. Moved it from my Downloads folder to the directory in WSL (from WSL) and unextracted that with no joy. Interestingly enough, dist-upgrade doesn't work either for the same reason. The fix *supposedly* is coming in an upcoming Insider build but didn't seem to make this latest one released. I'm half tempted to do a repair on the app for Ubuntu.exe to reset it without uninstalling (still requires launching from the Windows Store after repair, otherwise will open an OpenSuse console) and creating a Docker container for the Insider build that contains the fix. It would be great if I could run the Docker engine in WSL but that isn't on the radar right now so Docker for Windows unfortunately for now. Part of the problem I know is that repositories are missing and you'd have to know what needs to be held before performing the upgrade and the only way to successfully install in the first place is to install immediately after doing the apt-gets to update everything and then upgrade to 17.10. It's really quite annoying seeing that in a dev environment, you're always testing something new and not being able to properly utilize 17.10 is a pain. I'm probably just going to have to wait until 18.04 LTS is released in addition to the next Windows upgrade in March for things to start rolling.
